internal/transport/http2_server.go

```diff
@@ -1324,7 +1341,13 @@ func (t *http2Server) closeStream(s *ServerStream, rst bool, rstCode http2.ErrCo
 	// called to interrupt the potential blocking on other goroutines.
 	s.cancel()
 
-	s.swapState(streamDone)
+	oldState := s.swapState(streamDone)
+	if oldState == streamDone {
```

Comment:
Was this a bug before?  Or did we have something that prevented it?  Or was the code tolerant of multiple calls?